src/main/java/uk/ac/ebi/ega/egafuse/config/EgaFuseApplicationConfig.java

```diff
@@ -62,11 +63,12 @@ public void setFileDatabaseService(boolean isTreeStructure) {
     }
 
     @Bean
-    public AsyncLoadingCache<CacheKey, byte[]> cache(@Value("${maxCache}") int MAX_CACHE_SIZE,
+    public AsyncLoadingCache<CacheKey, byte[]> cache(@Value("${maxCache}") int MAX_CACHE_SIZE, @Value("${connectionPerFile}") int cachePrefetch,
             IFileChunkDownloadService fileChunkDownloadService) {
         return  Caffeine.newBuilder()
                         .expireAfterWrite(5, TimeUnit.HOURS)
                         .maximumSize(MAX_CACHE_SIZE)
+                        .executor(Executors.newFixedThreadPool(cachePrefetch))

```

Comment:
Technically the size of the executor should be max connections. Remember that we can prefetch N chunks but the user can still try to access more files.